axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/StockMoveServiceSupplychainImpl.java

```diff
@@ -160,36 +161,39 @@ public String realizeStockMove(StockMove stockMove, boolean check) throws Axelor
     String newStockSeq = super.realizeStockMove(stockMove, check);
     AppSupplychain appSupplychain = appSupplyChainService.getAppSupplychain();
 
-    if (stockMove.getSaleOrder() != null) {
-      updateSaleOrderLinesDeliveryState(stockMove, !stockMove.getIsReversion());
-      // Update linked saleOrder delivery state depending on BackOrder's existence
-      SaleOrder saleOrder = stockMove.getSaleOrder();
-      if (newStockSeq != null) {
-        saleOrder.setDeliveryState(SaleOrderRepository.DELIVERY_STATE_PARTIALLY_DELIVERED);
-      } else {
-        Beans.get(SaleOrderStockService.class).updateDeliveryState(saleOrder);
-
-        if (appSupplychain.getTerminateSaleOrderOnDelivery()) {
-          terminateOrConfirmSaleOrderStatus(saleOrder);
+    Set<SaleOrder> saleOrderSet = stockMove.getSaleOrderSet();
+    if (ObjectUtils.notEmpty(saleOrderSet)) {
+      for (SaleOrder saleOrder : saleOrderSet) {
+        updateSaleOrderLinesDeliveryState(stockMove, !stockMove.getIsReversion());
+        // Update linked saleOrder delivery state depending on BackOrder's existence
+        if (newStockSeq != null) {
+          saleOrder.setDeliveryState(SaleOrderRepository.DELIVERY_STATE_PARTIALLY_DELIVERED);
+        } else {
+          Beans.get(SaleOrderStockService.class).updateDeliveryState(saleOrder);
```

Comment:
Hello, please replace Beans.get by injection by constructor, especially now since we are in a for each